--- /src/main/java/org/apache/commons/cli/TypeHandler.java
+++ /src/main/java/org/apache/commons/cli/TypeHandler.java
@@ -89,7 +89,7 @@
         }
         else if (PatternOptionBuilder.EXISTING_FILE_VALUE == clazz)
         {
+            return openFile(str);
-            return createFile(str);
         }
         else if (PatternOptionBuilder.FILES_VALUE == clazz)
         {
@@ -231,6 +231,17 @@
      * @return The file input stream represented by <code>str</code>.
      * @throws ParseException if the file is not exist or not readable
      */
+    public static FileInputStream openFile(String str) throws ParseException
+    {
+        try
+        {
+            return new FileInputStream(str);
+        }
+        catch (FileNotFoundException e)
+        {
+            throw new ParseException("Unable to find file: " + str);
+        }
+    }
 
     /**
      * Returns the File[] represented by <code>str</code>.
