--- /src/com/google/javascript/rhino/jstype/FunctionType.java
+++ /src/com/google/javascript/rhino/jstype/FunctionType.java
@@ -868,7 +868,10 @@
     // Warning about typeOfThis if it doesn't resolve to an ObjectType
     // is handled further upstream.
     // TODO(nicksantos): Handle this correctly if we have a UnionType.
+    JSType maybeTypeOfThis = safeResolve(typeOfThis, t, scope);
+    if (maybeTypeOfThis instanceof ObjectType) {
+      typeOfThis = (ObjectType) maybeTypeOfThis;
+    }
-    typeOfThis = (ObjectType) safeResolve(typeOfThis, t, scope);
 
     boolean changed = false;
     ImmutableList.Builder<ObjectType> resolvedInterfaces =
