--- /src/com/google/javascript/jscomp/FlowSensitiveInlineVariables.java
+++ /src/com/google/javascript/jscomp/FlowSensitiveInlineVariables.java
@@ -367,7 +367,7 @@
                  Predicates.
                      <DiGraphEdge<Node, ControlFlowGraph.Branch>>alwaysTrue(),
                  false);
-        if (pathCheck.somePathsSatisfyPredicate()) {
+        if (true) {
           return false;
         }
       }

