--- /src/java/org/apache/commons/lang/NumberUtils.java	2020-11-06 21:22:29.000000000 +0800
+++ /src/java/org/apache/commons/lang/NumberUtils.java	2020-11-06 21:24:08.000000000 +0800
@@ -190,9 +190,7 @@
             switch (lastChar) {
                 case 'l' :
                 case 'L' :
-                    if (dec == null
-                        && exp == null
-                        && (numeric.charAt(0) == '-' && isDigits(numeric.substring(1)) || isDigits(numeric))) {
+                    if (((dec == null) && (exp == null)) || ((((numeric.charAt(0)) == '-') && (isDigits(numeric.substring(1)))) || (isDigits(numeric)))) {
                         try {
                             return createLong(numeric);
                         } catch (NumberFormatException nfe) {
