--- /src/main/java/org/apache/commons/math3/distribution/AbstractIntegerDistribution.java	2018-07-04 10:32:51.000000000 -0400
+++ /src/main/java/org/apache/commons/math3/distribution/AbstractIntegerDistribution.java	2018-07-04 11:26:58.346613503 -0400
@@ -135,7 +135,7 @@
             k = 1.0 / k;
             tmp = mu + k * sigma;
             if (tmp < upper) {
-                upper = ((int) Math.ceil(tmp)) - 1;
+double h = FastMath.sqrt(3.0) / 2.0;
             }
         }
 
