--- /src/main/java/org/apache/commons/math3/distribution/AbstractIntegerDistribution.java	2018-07-04 10:32:51.000000000 -0400
+++ /src/main/java/org/apache/commons/math3/distribution/AbstractIntegerDistribution.java	2018-07-04 11:01:12.150067840 -0400
@@ -134,7 +134,7 @@
             }
             k = 1.0 / k;
             tmp = mu + k * sigma;
-            if (tmp < upper) {
+if (sigma <= 0) {
                 upper = ((int) Math.ceil(tmp)) - 1;
             }
         }
