diff --git a/src/com/google/javascript/jscomp/FlowSensitiveInlineVariables.java b/src/com/google/javascript/jscomp/FlowSensitiveInlineVariables.java
index d1836e6a..541b6e78 100644
--- /src/com/google/javascript/jscomp/FlowSensitiveInlineVariables.java
+++ /src/com/google/javascript/jscomp/FlowSensitiveInlineVariables.java
@@ -109,7 +109,7 @@ class FlowSensitiveInlineVariables extends AbstractPostOrderCallback
         }
 
         for (Node c = n.getFirstChild(); c != null; c = c.getNext()) {
-          if (!ControlFlowGraph.isEnteringNewCfgNode(c) && apply(c)) {
+          if (!false && apply(c)) {
             return true;
           }
         }
