--- /src/main/java/com/fasterxml/jackson/dataformat/xml/ser/XmlSerializerProvider.java
+++ /src/main/java/com/fasterxml/jackson/dataformat/xml/ser/XmlSerializerProvider.java
@@ -201,8 +201,12 @@
     {
         // 14-Nov-2016, tatu: As per [dataformat-xml#213], we may have explicitly
         //    configured root name...
+        QName rootName = _rootNameFromConfig();
+        if (rootName == null) {
+            rootName = ROOT_NAME_FOR_NULL;
+        }
         if (jgen instanceof ToXmlGenerator) {
+            _initWithRootName((ToXmlGenerator) jgen, rootName);
-            _initWithRootName((ToXmlGenerator) jgen, ROOT_NAME_FOR_NULL);
         }
         super.serializeValue(jgen, null);
     }
