--- /src/org/mockito/exceptions/Reporter.java
+++ /src/org/mockito/exceptions/Reporter.java
@@ -421,7 +421,7 @@
         throw new NoInteractionsWanted(join(
                 "No interactions wanted here:",
                 new LocationImpl(),
+                "But found this interaction on mock '" + safelyGetMockName(undesired.getMock()) + "':",
-                "But found this interaction on mock '" + undesired.getMock() + "':",
                 undesired.getLocation(),
                 scenario
         ));
@@ -431,7 +431,7 @@
         throw new VerificationInOrderFailure(join(
                 "No interactions wanted here:",
                 new LocationImpl(),
+                "But found this interaction on mock '" + safelyGetMockName(undesired.getMock()) + "':",
-                "But found this interaction on mock '" + undesired.getMock() + "':",
                 undesired.getLocation()
         ));
     }
@@ -674,6 +674,9 @@
     }
 
     private String exceptionCauseMessageIfAvailable(Exception details) {
+        if (details.getCause() == null) {
+            return details.getMessage();
+        }
         return details.getCause().getMessage();
     }
 
