--- /src/com/google/javascript/jscomp/FunctionTypeBuilder.java
+++ /src/com/google/javascript/jscomp/FunctionTypeBuilder.java
@@ -181,7 +181,7 @@
       // create interfaces JSType, ObjectType, FunctionType etc and have
       // separate implementation instead of the class hierarchy, so that
       // union types can also be object types, etc.
+      if (!type.restrictByNotNullOrUndefined().isSubtype(
-      if (!type.isSubtype(
               typeRegistry.getNativeType(OBJECT_TYPE))) {
         reportWarning(THIS_TYPE_NON_OBJECT, type.toString());
         return false;
--- /src/com/google/javascript/rhino/jstype/FunctionType.java
+++ /src/com/google/javascript/rhino/jstype/FunctionType.java
@@ -877,6 +877,9 @@
     // mean "nullable Foo". For certain tags (like @extends) we de-nullify
     // the name for them.
     JSType maybeTypeOfThis = safeResolve(typeOfThis, t, scope);
+    if (maybeTypeOfThis != null) {
+      maybeTypeOfThis = maybeTypeOfThis.restrictByNotNullOrUndefined();
+    }
     if (maybeTypeOfThis instanceof ObjectType) {
       typeOfThis = (ObjectType) maybeTypeOfThis;
     }
