--- /src/com/google/javascript/jscomp/type/ChainableReverseAbstractInterpreter.java
+++ /src/com/google/javascript/jscomp/type/ChainableReverseAbstractInterpreter.java
@@ -610,9 +610,13 @@
     public JSType caseObjectType(ObjectType type) {
       if (value.equals("function")) {
         JSType ctorType = getNativeType(U2U_CONSTRUCTOR_TYPE);
+        if (resultEqualsValue) {
-        return resultEqualsValue && ctorType.isSubtype(type) ? ctorType : null;
           // Objects are restricted to "Function", subtypes are left
+          return ctorType.getGreatestSubtype(type);
+        } else {
           // Only filter out subtypes of "function"
+          return type.isSubtype(ctorType) ? null : type;
+        }
       }
       return matchesExpectation("object") ? type : null;
     }
