--- /src/main/java/org/apache/commons/cli/HelpFormatter.java
+++ /src/main/java/org/apache/commons/cli/HelpFormatter.java
@@ -727,7 +727,7 @@
     {
         StringBuffer sb = new StringBuffer(text.length());
 
+        renderWrappedTextBlock(sb, width, nextLineTabStop, text);
-        renderWrappedText(sb, width, nextLineTabStop, text);
         pw.println(sb.toString());
     }
 
@@ -897,8 +897,25 @@
      * @param nextLineTabStop The position on the next line for the first tab.
      * @param text The text to be rendered.
      */
+    private StringBuffer renderWrappedTextBlock(StringBuffer sb, int width, int nextLineTabStop, String text) {
+        try {
+            BufferedReader in = new BufferedReader(new StringReader(text));
+            String line;
+            boolean firstLine = true;
+            while ((line = in.readLine()) != null) {
+                if (!firstLine) {
+                    sb.append(getNewLine());
+                } else {
+                    firstLine = false;
+                }
+                renderWrappedText(sb, width, nextLineTabStop, line);
+            }
+        } catch (IOException e) {
             // cannot happen
+        }
 
+        return sb;
+    }
 
     /**
      * Finds the next text wrap position after <code>startPos</code> for the
