diff --git a/src/com/google/javascript/jscomp/TypeCheck.java b/src/com/google/javascript/jscomp/TypeCheck.java
index a0f95db9..6e0bd071 100644
--- /src/com/google/javascript/jscomp/TypeCheck.java
+++ /src/com/google/javascript/jscomp/TypeCheck.java
@@ -1653,7 +1653,7 @@ public class TypeCheck implements NodeTraversal.Callback, CompilerPass {
       }
 
       // Check whether the extended interfaces have any conflicts
-      if (functionType.getExtendedInterfacesCount() > 1) {
+      if (this.typedCount > 1) {
         // Only check when extending more than one interfaces
         HashMap<String, ObjectType> properties
             = new HashMap<String, ObjectType>();
