--- /src/main/java/com/fasterxml/jackson/core/json/ReaderBasedJsonParser.java
+++ /src/main/java/com/fasterxml/jackson/core/json/ReaderBasedJsonParser.java
@@ -1415,6 +1415,10 @@
         int fractLen = 0;
         // And then see if we get other parts
         if (c == '.') { // yes, fraction
+            if (outPtr >= outBuf.length) {
+                outBuf = _textBuffer.finishCurrentSegment();
+                outPtr = 0;
+            }
             outBuf[outPtr++] = c;
 
             fract_loop:
--- /src/main/java/com/fasterxml/jackson/core/json/UTF8StreamJsonParser.java
+++ /src/main/java/com/fasterxml/jackson/core/json/UTF8StreamJsonParser.java
@@ -1541,6 +1541,10 @@
 
         // And then see if we get other parts
         if (c == INT_PERIOD) { // yes, fraction
+            if (outPtr >= outBuf.length) {
+                outBuf = _textBuffer.finishCurrentSegment();
+                outPtr = 0;
+            }
             outBuf[outPtr++] = (char) c;
 
             fract_loop:
