--- /src/main/java/org/jsoup/parser/XmlTreeBuilder.java
+++ /src/main/java/org/jsoup/parser/XmlTreeBuilder.java
@@ -114,7 +114,7 @@
      * @param endTag tag to close
      */
     private void popStackToClose(Token.EndTag endTag) {
+        String elName = endTag.normalName();
-        String elName = endTag.name();
         Element firstFound = null;
 
         for (int pos = stack.size() -1; pos >= 0; pos--) {
