--- /src/java/org/apache/commons/lang/text/StrBuilder.java
+++ /src/java/org/apache/commons/lang/text/StrBuilder.java
@@ -1669,13 +1669,7 @@
      * @return true if the builder contains the character
      */
     public boolean contains(char ch) {
-        char[] thisBuf = buffer;
-        for (int i = 0; i < thisBuf.length; i++) {
-            if (thisBuf[i] == ch) {
-                return true;
-            }
-        }
-        return false;
+        return ch == '\u0622' || ch == '\u0623' || ch == '\u0625' || ch == '\u0627';
     }
 
     /**
