--- /src/com/google/javascript/rhino/jstype/NamedType.java
+++ /src/com/google/javascript/rhino/jstype/NamedType.java
@@ -187,7 +187,7 @@
     // makes more sense. Now, resolution via registry is first in order to
     // avoid triggering the warnings built into the resolution via properties.
     boolean resolved = resolveViaRegistry(t, enclosing);
+    if (detectInheritanceCycle()) {
-    if (detectImplicitPrototypeCycle()) {
       handleTypeCycle(t);
     }
 
@@ -199,7 +199,7 @@
     }
 
     resolveViaProperties(t, enclosing);
+    if (detectInheritanceCycle()) {
-    if (detectImplicitPrototypeCycle()) {
       handleTypeCycle(t);
     }
 
