--- /src/main/java/com/fasterxml/jackson/databind/deser/DefaultDeserializationContext.java
+++ /src/main/java/com/fasterxml/jackson/databind/deser/DefaultDeserializationContext.java
@@ -84,6 +84,9 @@
         /* 02-Apr-2015, tatu: As per [databind#742] should allow 'null', similar to how
          *   missing id already works.
          */
+        if (id == null) {
+            return null;
+        }
 
         final ObjectIdGenerator.IdKey key = gen.key(id);
 
--- /src/main/java/com/fasterxml/jackson/databind/deser/impl/ObjectIdValueProperty.java
+++ /src/main/java/com/fasterxml/jackson/databind/deser/impl/ObjectIdValueProperty.java
@@ -89,6 +89,9 @@
          *  will be generated externally, at a later point, and is not available
          *  quite yet. Typical use case is with DB inserts.
          */
+        if (id == null) {
+            return null;
+        }
         
         ReadableObjectId roid = ctxt.findObjectId(id, _objectIdReader.generator, _objectIdReader.resolver);
         roid.bindItem(instance);
