--- /src/main/java/org/apache/commons/compress/compressors/bzip2/BZip2CompressorInputStream.java
+++ /src/main/java/org/apache/commons/compress/compressors/bzip2/BZip2CompressorInputStream.java
@@ -165,6 +165,9 @@
         if (this.in == null) {
             throw new IOException("stream closed");
         }
+        if (len == 0) {
+            return 0;
+        }
 
         final int hi = offs + len;
         int destOffs = offs;
