--- /src/com/google/javascript/jscomp/CheckGlobalThis.java
+++ /src/com/google/javascript/jscomp/CheckGlobalThis.java
@@ -105,7 +105,8 @@
       if (!(pType == Token.BLOCK ||
             pType == Token.SCRIPT ||
             pType == Token.NAME ||
+            pType == Token.ASSIGN ||
+            pType == Token.OBJECTLIT)) {
-            pType == Token.ASSIGN)) {
         return false;
       }
     }
--- /src/com/google/javascript/jscomp/RuntimeTypeCheck.java
+++ /src/com/google/javascript/jscomp/RuntimeTypeCheck.java
@@ -244,6 +244,11 @@
 
       // To satisfy normalization constraints, the type checking must be
       // added after any inner function declarations.
+      for (Node next = block.getFirstChild();
+           next != null && NodeUtil.isFunctionDeclaration(next);
+           next = next.getNext()) {
+        insertionPoint = next;
+      }
 
       for (Node paramType : funType.getParameters()) {
         // Can this ever happen?
