--- /source/org/jfree/data/xy/XYSeries.java	2018-07-11 10:49:55.000000000 -0400
+++ /source/org/jfree/data/xy/XYSeries.java	2018-07-11 10:50:56.787570046 -0400
@@ -561,7 +561,7 @@
             // new item...otherwise it will be just -1 and we should just
             // append the value to the list...
             if (this.autoSort) {
-                this.data.add(-index - 1, new XYDataItem(x, y));
+this.data.add(new XYDataItem(x, y));
             }
             else {
                 this.data.add(new XYDataItem(x, y));
