--- /src/com/google/javascript/jscomp/CollapseProperties.java
+++ /src/com/google/javascript/jscomp/CollapseProperties.java
@@ -481,6 +481,9 @@
     Node greatGramps = gramps.getParent();
     Node greatGreatGramps = greatGramps.getParent();
 
+    if (rvalue != null && rvalue.getType() == Token.FUNCTION) {
+      checkForHosedThisReferences(rvalue, refName.docInfo, refName);
+    }
 
     // Create the new alias node.
     Node nameNode = NodeUtil.newName(
--- /src/com/google/javascript/jscomp/GlobalNamespace.java
+++ /src/com/google/javascript/jscomp/GlobalNamespace.java
@@ -917,7 +917,7 @@
       }
 
       // If this is aliased, then its properties can't be collapsed either.
+      if (aliasingGets > 0) {
-      if (type != Type.FUNCTION && aliasingGets > 0) {
         return false;
       }
 
