--- /src/com/google/javascript/jscomp/CodeConsumer.java
+++ /src/com/google/javascript/jscomp/CodeConsumer.java
@@ -238,7 +238,7 @@
       add(" ");
     }
 
+    if ((long) x == x && !isNegativeZero(x)) {
-    if ((long) x == x) {
       long value = (long) x;
       long mantissa = value;
       int exp = 0;
@@ -256,7 +256,10 @@
     } else {
       add(String.valueOf(x));
     }
+  }
 
+  static boolean isNegativeZero(double x) {
+    return x == 0.0 && Math.copySign(1, x) == -1.0;
   }
 
   static boolean isWordChar(char ch) {
