--- /src/main/java/org/apache/commons/cli/OptionGroup.java
+++ /src/main/java/org/apache/commons/cli/OptionGroup.java
@@ -19,8 +19,8 @@
 
 import java.io.Serializable;
 import java.util.Collection;
-import java.util.HashMap;
 import java.util.Iterator;
+import java.util.LinkedHashMap;
 import java.util.Map;
 
 /**
@@ -34,7 +34,7 @@
     private static final long serialVersionUID = 1L;
     
     /** hold the options */
+    private final Map<String, Option> optionMap = new LinkedHashMap<String, Option>();
-    private final Map<String, Option> optionMap = new HashMap<String, Option>();
 
     /** the name of the selected option */
     private String selected;
--- /src/main/java/org/apache/commons/cli/Options.java
+++ /src/main/java/org/apache/commons/cli/Options.java
@@ -21,7 +21,6 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.HashMap;
 import java.util.HashSet;
 import java.util.LinkedHashMap;
 import java.util.List;
@@ -58,7 +57,7 @@
     private final List<Object> requiredOpts = new ArrayList<Object>();
 
     /** a map of the option groups */
+    private final Map<String, OptionGroup> optionGroups = new LinkedHashMap<String, OptionGroup>();
-    private final Map<String, OptionGroup> optionGroups = new HashMap<String, OptionGroup>();
 
     /**
      * Add the specified option group.
