diff --git a/src/com/google/javascript/rhino/Node.java b/src/com/google/javascript/rhino/Node.java
index 9e9d8463..e66c8368 100644
--- /src/com/google/javascript/rhino/Node.java
+++ /src/com/google/javascript/rhino/Node.java
@@ -1453,7 +1453,7 @@ public class Node implements Cloneable, Serializable {
         throw new IllegalStateException();
       }
       res = n.checkTreeEqualsImpl(n2);
-      if (res != null) {
+      if (res == null) {
         return res;
       }
     }
