--- /src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializerFactory.java
+++ /src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializerFactory.java
@@ -492,7 +492,7 @@
             boolean ignoreAny = ignorals.getIgnoreUnknown();
             builder.setIgnoreUnknownProperties(ignoreAny);
             // Or explicit/implicit definitions?
+            ignored = ignorals.findIgnoredForDeserialization();
-            ignored = ignorals.getIgnored();
             for (String propName : ignored) {
                 builder.addIgnorable(propName);
             }
