--- /src/com/google/javascript/jscomp/GlobalNamespace.java
+++ /src/com/google/javascript/jscomp/GlobalNamespace.java
@@ -904,6 +904,10 @@
       // Don't try to collapse if the one global set is a twin reference.
       // We could theoretically handle this case in CollapseProperties, but
       // it's probably not worth the effort.
+      Preconditions.checkNotNull(declaration);
+      if (declaration.getTwin() != null) {
+        return false;
+      }
 
       if (isClassOrEnum) {
         return true;
--- /src/com/google/javascript/rhino/JSDocInfoBuilder.java
+++ /src/com/google/javascript/rhino/JSDocInfoBuilder.java
@@ -186,9 +186,7 @@
    * @return {@code true} if the description was recorded.
    */
   public boolean recordBlockDescription(String description) {
-    if (parseDocumentation) {
     populated = true;
-    }
     return currentInfo.documentBlock(description);
   }
 
