--- /src/com/google/javascript/jscomp/TypedScopeCreator.java
+++ /src/com/google/javascript/jscomp/TypedScopeCreator.java
@@ -1518,11 +1518,14 @@
             || FunctionTypeBuilder.isFunctionTypeDeclaration(info));
       }
 
+      if (inferred && rhsValue != null && rhsValue.isFunction()) {
-      if (inferred) {
         // Determining declaration for #2
+        if (info != null) {
+          inferred = false;
+        } else if (!scope.isDeclared(qName, false) &&
+                   n.isUnscopedQualifiedName()) {
+          inferred = false;
+        }
-        inferred = !(rhsValue != null &&
-            rhsValue.isFunction() &&
-            (info != null || !scope.isDeclared(qName, false)));
       }
 
       if (!inferred) {
