--- /src/java/org/apache/commons/jxpath/ri/axes/UnionContext.java
+++ /src/java/org/apache/commons/jxpath/ri/axes/UnionContext.java
@@ -17,8 +17,10 @@
 package org.apache.commons.jxpath.ri.axes;
 
 import java.util.ArrayList;
+import java.util.Iterator;
 
 import org.apache.commons.jxpath.BasicNodeSet;
+import org.apache.commons.jxpath.Pointer;
 import org.apache.commons.jxpath.ri.EvalContext;
 import org.apache.commons.jxpath.ri.model.NodePointer;
 
@@ -53,13 +55,17 @@
                     while (ctx.nextNode()) {
                         NodePointer ptr = ctx.getCurrentNodePointer();
                         if (!pointers.contains(ptr)) {
-                            nodeSet.add(ptr);
                             pointers.add(ptr);
                         }
                     }
                 }
             }
+            sortPointers(pointers);
+
+            for (Iterator it = pointers.iterator(); it.hasNext();) {
+                nodeSet.add((Pointer) it.next());
+            }
         }
         return super.setPosition(position);
     }
+}
\ No newline at end of file
-}
