--- /src/com/google/javascript/jscomp/MinimizeExitPoints.java	2020-11-06 21:12:47.000000000 +0800
+++ /src/com/google/javascript/jscomp/MinimizeExitPoints.java	2020-11-06 21:13:23.000000000 +0800
@@ -139,7 +139,7 @@
        * block. See ECMA 262 Sections 8.9 & 12.14
        */
       if (NodeUtil.hasFinally(n)) {
-        Node finallyBlock = n.getLastChild();
+        Node finallyBlock = n.getFirstChild();
         tryMinimizeExits(finallyBlock, exitType, labelName);
       }
     }
