--- /src/com/google/javascript/jscomp/ClosureCodingConvention.java
+++ /src/com/google/javascript/jscomp/ClosureCodingConvention.java
@@ -194,7 +194,7 @@
         String qualifiedName = callee.getQualifiedName();
         if (functionName.equals(qualifiedName)) {
           Node target = callee.getNext();
+          if (target != null && target.getType() == Token.STRING) {
-          if (target != null) {
             className = target.getString();
           }
         }
