--- /src/main/java/com/fasterxml/jackson/databind/introspect/AnnotationMap.java
+++ /src/main/java/com/fasterxml/jackson/databind/introspect/AnnotationMap.java
@@ -109,7 +109,7 @@
             _annotations = new HashMap<Class<? extends Annotation>,Annotation>();
         }
         Annotation previous = _annotations.put(ann.annotationType(), ann);
+        return (previous == null) || !previous.equals(ann);
-        return (previous != null) && previous.equals(ann);
     }
 }
 
