diff --git a/src/main/java/org/apache/commons/math3/linear/OpenMapRealVector.java b/src/main/java/org/apache/commons/math3/linear/OpenMapRealVector.java
index 821482744..2c99edf1d 100644
--- /src/main/java/org/apache/commons/math3/linear/OpenMapRealVector.java
+++ /src/main/java/org/apache/commons/math3/linear/OpenMapRealVector.java
@@ -136,9 +136,7 @@ public class OpenMapRealVector extends SparseRealVector
         this.epsilon = epsilon;
         for (int key = 0; key < values.length; key++) {
             double value = values[key];
-            if (!isDefaultValue(value)) {
-                entries.put(key, value);
-            }
+            entries.put(key, value);
         }
     }
 
