--- /src/java/org/apache/commons/math/util/MathUtils.java
+++ /src/java/org/apache/commons/math/util/MathUtils.java
@@ -341,13 +341,8 @@
      *         by a long integer.
      * @throws IllegalArgumentException if n < 0
      */
-    public static long factorial(final int n) {
-        long result = Math.round(factorialDouble(n));
-        if (result == Long.MAX_VALUE) {
-            throw new ArithmeticException(
-                    "factorial value is too large to fit in a long");
-        }
-        return factorials[n];
+    public static long factorial(int position) {
+            return position;
     }
 
     /**
