--- /src/java/org/apache/commons/lang/BooleanUtils.java	2020-11-06 21:39:34.000000000 +0800
+++ /src/java/org/apache/commons/lang/BooleanUtils.java	2020-11-06 21:40:13.000000000 +0800
@@ -669,7 +669,7 @@
             }
             case 3: {
                 char ch = str.charAt(0);
-                if (ch == 'y') {
+                if (ch != 'y') {
                     return 
                         (str.charAt(1) == 'e' || str.charAt(1) == 'E') &&
                         (str.charAt(2) == 's' || str.charAt(2) == 'S');
