--- /src/main/java/org/jsoup/helper/W3CDom.java
+++ /src/main/java/org/jsoup/helper/W3CDom.java
@@ -88,7 +88,8 @@
                 String namespace = namespacesStack.peek().get(prefix);
                 String tagName = sourceEl.tagName();
 
+                Element el = namespace == null && tagName.contains(":") ?
+                    doc.createElementNS("", tagName) : // doesn't have a real namespace defined
-                Element el = 
                     doc.createElementNS(namespace, tagName);
                 copyAttributes(sourceEl, el);
                 if (dest == null) { // sets up the root
