--- /src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java
+++ /src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java
@@ -498,11 +498,11 @@
             } else if ("linkpath".equals(key)){
                 currEntry.setLinkName(val);
             } else if ("gid".equals(key)){
+                currEntry.setGroupId(Long.parseLong(val));
-                currEntry.setGroupId(Integer.parseInt(val));
             } else if ("gname".equals(key)){
                 currEntry.setGroupName(val);
             } else if ("uid".equals(key)){
+                currEntry.setUserId(Long.parseLong(val));
-                currEntry.setUserId(Integer.parseInt(val));
             } else if ("uname".equals(key)){
                 currEntry.setUserName(val);
             } else if ("size".equals(key)){
