--- /src/main/java/org/jsoup/helper/StringUtil.java
+++ /src/main/java/org/jsoup/helper/StringUtil.java
@@ -128,7 +128,10 @@
         // 160 is &nbsp; (non-breaking space). Not in the spec but expected.
     }
 
+    public static boolean isInvisibleChar(int c) {
+        return Character.getType(c) == 16 && (c == 8203 || c == 8204 || c == 8205 || c == 173);
         // zero width sp, zw non join, zw join, soft hyphen
+    }
 
     /**
      * Normalise the whitespace within this string; multiple spaces collapse to a single, and all whitespace characters
@@ -162,7 +165,7 @@
                 accum.append(' ');
                 lastWasWhite = true;
             }
+            else if (!isInvisibleChar(c)) {
-            else {
                 accum.appendCodePoint(c);
                 lastWasWhite = false;
                 reachedNonWhite = true;
